lib/braintree/gateway.js

```diff
@@ -90,10 +91,11 @@ class Gateway {
 
   pagingFunctionGenerator(search, url, SubjectType, pagedResultsKey, getSubject) {
     return (ids, callback) => {
-      search.ids().in(ids);
       let gateway = this.gateway;
+      let idsSearch = new TransactionSearch();
```

Comment:
This paging function is used for all searches, not just transaction searches, so I don't think this works.

Reply:
In that case could you suggest what do you think for below approaches ? 

1. Checking for presence of `ids` and fallback to this approach and in all other cases proceed with original `search` reference
1. We have `totalItems` sent back by api to us at this point we can relay the same to `search_response_stream.js` and flush the `bufferedResults` based over it

Reply:
Thanks @crookedneighbor for looking into this  but `pagingFunctionGenerator` callback is it not supposed to collect `ids` from other filters to go to `advanced_search` with only `ids` maybe. 

None the less the alternate approach that I can think of is to modify `TransactionSearch` to have a flag for controlling this behaviour. What do you think about this ?

Reply:
@crookedneighbor as an alternate approach controlling the ids override internally with flag consumers can enable with additional options captured during setting up search.